lib/view/repositories_screen.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../core/http/api_response.dart';
+import '../models/get_repositories/repository_rto.dart';
+import '../view_model/repositories_view_model.dart';
+import 'widget/app_error_widget.dart';
+import 'widget/drop_down.dart';
+import 'widget/loading_widget.dart';
+import 'widget/repository_item_widget.dart';
+
+class RepositoriesScreen extends StatefulWidget {
+  static const routeName = "repositories_screen";
+  const RepositoriesScreen({super.key});
+
+  @override
+  State<RepositoriesScreen> createState() => _RepositoriesScreenState();
+}
+
+class _RepositoriesScreenState extends State<RepositoriesScreen> {
+  final RepositoriesViewModel viewModel = RepositoriesViewModel();
+  final _languages = const <String>[
+    "Dart",
+    "Java",
+    "JavaScript",
+    "Kotlin",
+    "PHP",
+    "Python",
+    "Ruby",
+    "Swift",
+    "TypeScript",
+  ];
+  late String _selectedLanguage;
+
+  void _didSelect({required String item}) {
+    setState(() {
+      _selectedLanguage = item;
+      viewModel.getRepositories(_selectedLanguage);
+    });
+  }
+
+  @override
+  void initState() {
+    _selectedLanguage = _languages[0];
+    viewModel.getRepositories(_selectedLanguage);
+    super.initState();
+  }
+
+  Widget _getRepositoriesListView(List<RepositoryRTO> repositories) =>
+      ListView.builder(
+        scrollDirection: Axis.vertical,
+        shrinkWrap: true,
+        itemCount: repositories.length,
+        itemBuilder: (context, index) =>
+            RepositoryItemWidget(repositories[index]),
+      );
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
+        appBar: AppBar(
+          title: const Text("Repositories"),
+        ),
+        body: Padding(
+          padding: const EdgeInsets.symmetric(vertical: 5, horizontal: 10),
+          child: Column(
+            children: [
+              DropDown(
+                items: _languages,
+                didSelect: _didSelect,
+                dropdownValue: _selectedLanguage,
+              ),
+              const SizedBox(
+                height: 10,
+              ),
+              Expanded(
+                child: ChangeNotifierProvider<RepositoriesViewModel>(
```

Comment:
We tend to move ChangeNotifierProvider, Consumers and other widgets who provide or consumer data outside of our Widgets. By doing that, the widget itself becomes more readable. What you could do is:

Create a Widget called `Repositories`, being the public widget. And let that widget build:
- The ChangeNotifierProvider, providing the RepositoriesViewModel
- Consume the RepositoriesViewModel
- Pass the RepositoriesViewModel into the constructor of the `_RepositoriesScreen` widget. Just like the RepositoriesScreen widget you now have.
- Then from the widget build inside your _RepositoriesScreenState, you can use `widget.viewModel` to retrieve the viewModel and act.

Reply:
@Ramakrishna530 I think you misunderstood @RtgrV request. At least the latest code is not following his suggestion.

He suggested to you to move the provisioning of view model to one level above the RepositoryScreen. Then everybody below in the widget tree can consume it. This becomes more useful when you have a complex UI with bunch of different widgets having to use the same view model.

Reply:
done